tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -179,22 +188,25 @@ def __init__(self, test_context):
             logger=self.logger,
         )
 
-        # Used in the consumer groups tests
-        self._exceptions = [None] * self.num_groups
-
-        self._fgo_node = self.test_context.cluster.alloc(
-            ClusterSpec.simple_linux(1))
-        self.logger.debug(f"Allocated verifier node {self._fgo_node[0].name}")
-
     def setUp(self):
         self.s3_client.empty_bucket(self.s3_bucket_name)
         self.s3_client.create_bucket(self.s3_bucket_name)
         self.redpanda.start()
 
-        spec = TopicSpec(partition_count=5)
-        self.kafka_tools.create_topic(spec)
-        self.topic = spec.name
-        self.topics.append(spec)
+        if self.test_context.function_name == 'test_consumer_groups':
```

Comment:
Same  comment as in constructor -- avoid doing if statements based on test string names, prefer to refactor distinct setup logic into distinct functions and/or classes.